core/db_models/fields/EE_Foreign_Key_String_Field.php

```diff
@@ -2,6 +2,28 @@
 
 class EE_Foreign_Key_String_Field extends EE_Foreign_Key_Field_Base
 {
+    /**
+     * Whether the value should be converted to uppercase on insertion.
+     *
+     * @var boolean
+     */
+    protected $is_uppercase;
+
+
+    /**
+     * @param string          $table_column  name of column for field
+     * @param string          $nicename      should be internationalized with __('blah','event_espresso')
+     * @param boolean         $nullable
+     * @param int|string      $default_value data type should match field type
+     * @param string|string[] $model_name    eg 'Event','Answer','Term', etc.
+     *                                       Basically its the model class's name without the "EEM_"
+     * @param boolean         $is_uppercase  Whether the value should be converted to uppercase on insertion.
+     */
+    public function __construct($table_column, $nicename, $nullable, $default_value, $model_name, $is_uppercase = true)
+    {
+        $this->is_uppercase = $is_uppercase;
+        parent::__construct($table_column, $nicename, $nullable, $default_value, $model_name);
+    }
```

Comment:
hmmm... this is actually not bad at all in terms of a solution for this issue because it does tackle it at the root...

will think about this more because it still seems like this uppercase rule is specific to the Country model and not `EE_Foreign_Key_String_Field`

Reply:
I wanted the change to be fully backward compatible without the need to change any older code, you never know where things can explode 💣

Reply:
yup... good thinking 💯